platforms/native/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/binaryinfo/NMToolFixture.groovy

```diff
@@ -42,16 +42,39 @@ class NMToolFixture {
     List<BinaryInfo.Symbol> listSymbols(File binaryFile) {
         def process = (findExe('nm') + ['-a', '-f', 'posix', binaryFile.absolutePath]).execute(environments, null)
         def lines = process.inputStream.readLines()
-        return lines.collect { line ->
+        def result = []
+        for (final line in lines) {
             // Looks like on Linux:
             // _main t 0 0
             //
             // Looks like on Windows (MinGW):
             // main T 0000000000401550
-            def splits = line.split(' ')
-            String name = splits[0]
-            char type = splits[1].getChars()[0]
-            return new BinaryInfo.Symbol(name, type, Character.isUpperCase(type))
+            def splits = line.split(' ').toList()
+
+            // The symbol name is the first element, but is not delimited so if it contains spaces, it will span multiple elements
```

Comment:
The format seems to have changed a little bit, so adjust the code for this. Also, handle symbols with spaces in their names.